### ISSUE TITLE
Problem With Separating Traffic Type for vmware host in Cloudstack UI

### ISSUE THREAD

**Issue by ahmadamirahmadi007** [apache/cloudstack#6593]
##### ISSUE TYPE
 * Bug Report
##### COMPONENT NAME
API OR UI
##### CLOUDSTACK VERSION
CloudStack 4.17.0.1-snapshot.20220730
4.18
##### CONFIGURATION

Vmware Compute Host
3 Standard VirtualSwitch In vCenter
-----------------
vSwitch0 : Storage And Management 
vSwitch1 : Public
vSwitch2 : Guest
advanced network without security group

##### OS / ENVIRONMENT
Ubuntu 20.04 LTS

##### SUMMARY
i have problem with Separating Physical Network For All Type Of Traffic (mgmt,pub,guest). i encounter the following error by separating them in Cloudstack UI Wizard: 
"Unable to add the external cluster"
this is while when i put them in just one physical Interface and one vSwitch0 , everything is OK.
there is same problem with virtual Distributed Switch.  
##### STEPS TO REPRODUCE
log file link
https://gist.github.com/ahmadamirahmadi007/2c8d02ab213924ba806bae065b3d4377
![1](https://user-images.githubusercontent.com/104001728/182009938-7048dcde-6aa8-4d42-ac1b-aa7de93c9e4d.png)

##### EXPECTED RESULTS
i want to separate All Traffic Type in different vss or vds
##### ACTUAL RESULTS
i can not do that

**Comment by ahmadamirahmadi007** [comment]
i can't even choose vss or vds in physical network section in cloudstack UI.
![physical net](https://user-images.githubusercontent.com/104001728/182325231-0835c444-0c64-467b-8151-1f4929ede773.png)

**Comment by shwstppr** [comment]
@ahmadamirahmadi007 In case of different physical networks, server was failing to add your cluster while facing problem with wrong vlan,
```
2022-07-31 08:14:02,240 INFO  [c.c.h.v.m.VmwareManagerImpl] (qtp681094281-498:ctx-7b80359c ctx-12777883) (logid:0d78ed6c) Preparing network on host com.cloud.hypervisor.vmware.util.VmwareContext@36fe5de2 for vSwitch0,null,vmwaresvs
2022-07-31 08:14:02,260 WARN  [c.c.h.v.VmwareServerDiscoverer] (qtp681094281-498:ctx-7b80359c ctx-12777883) (logid:0d78ed6c) Unable to connect to Vmware vSphere server. service address: vcsa1.acs.lab.local. java.lang.NumberFormatException: For input string: "null"
```

This seems to be a bug in the UI which adds the wrong hypervisor traffic label. I've created PR #6612 to fix it.
Currently traffic label for your different physical networs will read like,
`vSwitch0,null,vmwaresvs`
Though they should be like the following if you are not using a VLAN:
`vSwitch0,,vmwaresvs`

You can update the VMware traffic label by going into the particular physical network and using the Update traffic label action from the top right list of actions
![Screenshot from 2022-08-04 14-51-56](https://user-images.githubusercontent.com/153340/182816639-e586c733-4abd-46a5-9ef0-58a68b698c87.png)

Regarding your second query, option to select switch type isn't present in the physical network form as these networks are not linked to a particular swtich type. Rather it is the traffic type - Management, Public, Guest, etc.
You may continue to add physical network using the form and then add the desired traffic type to it (Add Traffic type action in physical network). After adding the desired traffic type you will have to update the VMware traffic label using the Update traffic label action. Unfortunately there isn't a select UI here so you will have to manually add the label in format,

SWITCHNAME,VLAN,SWITCHTYPE
eg:
`vSwitch0,,vmwaresvs`
`vSwitch1,201,vmwaredvs`
`vSwitch2,untagged,vmwaredvs`

**Comment by shwstppr** [comment]
Fixed with #6612 
Please reopen if you face this again after the fix
